src/visualization/include/visualization/interactive_tree_to_graphviz.h

```diff
@@ -21,10 +21,50 @@
 
 #include "tree_to_graphviz.h"
 
+#include <fmt/ostream.h>
 #include <search/search_tree.h>
 
 namespace tacos::visualization {
 
+/// Print a help message to the logger.
+void print_interactive_help();
+
+enum class Mode {
+	NAVIGATE,
+	INSERT,
+	INSERT_AND_FOLLOW,
+};
+
+std::ostream &operator<<(std::ostream &os, const Mode &mode);
+
+namespace details {
+template <typename ActionT, typename NodeT>
+std::map<int, const NodeT *>
+create_selector_map(
+  const std::map<std::pair<search::RegionIndex, ActionT>, std::shared_ptr<NodeT>> &children,
+  const std::set<NodeT *> &                                                        parents = {})
+{
+	std::map<int, const NodeT *> selector_map;
+	int                          node_index = 0;
+	for (const auto &node : parents) {
+		selector_map[node_index] = node;
+		fmt::print("{}: Parent \033[37m{}\033[0m\n", node_index, *node);
+		node_index += 1;
+	}
+	for (const auto &[action, node] : children) {
+		selector_map[node_index] = node.get();
+		fmt::print("{}: \033[34m({}, {})\033[0m -> \033[37m{}\033[0m\n",
+		           node_index,
+		           action.first,
+		           action.second,
+		           *node);
+		node_index += 1;
+	}
+	return selector_map;
+}
```

Comment:
My main concern with this function is: It seems that it does two things, it prints options which are selectable by the user and returns a map required to implement the user-choice afterwards. I think the printing should happen outside this function

Reply:
Good point! I chose to do both in the same function, as we do not (and should not) store all information that we want to print in the map (e.g., the action that leads to a child). If we add this to the resulting map, then we would somehow need to insert a pseudo action for the parent nodes. I think this would make the code less readable. Also, I think this function is really only helpful for this particular use case.

Reply:
ok - I was assuming something like that. The only option to circumvent this (with additional effort) would be to parse the map after generation and create the output then, but I agree, it is pretty much single-purpose here.